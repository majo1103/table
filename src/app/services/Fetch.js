import React from "react";

export class Fetch extends React.Component {
  //should be service
  render() {
    return (
      <div>
        <label htmlFor="name-input">Name</label>
        <input id="name-input" />
        <button>Load Greeting</button>
        <div data-testid="greeting-text" />
      </div>
    );
  }
}
