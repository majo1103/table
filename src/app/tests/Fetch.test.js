import React from "react";
import {
  render,
  waitForElement,
  cleanup,
  fireEvent
} from "react-testing-library";
// this adds custom expect matchers
import "jest-dom/extend-expect";
// the mock lives in a __mocks__ directory
import axios from "axios";
import { Fetch } from "../services/Fetch";
import MockAdapter from "axios-mock-adapter";

let mock = new MockAdapter(axios);
afterEach(cleanup);
afterAll(() => mock.restore);

test("displays greeting when clicking Load Greeting", async () => {
  const url = "/greeting";

  mock.onGet(url).reply(200, {
    data: { greeting: "Hello there" }
  });

  const { getByText, getByTestId, container } = render(<Fetch />);
  // Act
  fireEvent.click(getByText("Load Greeting"));
  axios.get(url).then(function(response) {
    console.log(response.data);
  });
  // let's wait for our mocked `get` request promise to resolve
  // wait will wait until the callback doesn't throw an error
  const greetingTextNode = await waitForElement(() =>
    getByTestId("greeting-text")
  );
  // here's a custom matcher!
  expect(getByTestId("greeting-text")).toHaveTextContent("Hello there");
  // snapshots work great with regular DOM nodes!
  expect(container.firstChild).toMatchSnapshot();
});
