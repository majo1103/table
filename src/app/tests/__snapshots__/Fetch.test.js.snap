// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays greeting when clicking Load Greeting 1`] = `
<div>
  <label
    for="name-input"
  >
    Name
  </label>
  <input
    id="name-input"
  />
  <button>
    Load Greeting
  </button>
  <div
    data-testid="greeting-text"
  >
    Hello there
  </div>
</div>
`;
